internal/handlers/tigris/tigrisdb/query.go

```diff
@@ -95,25 +110,45 @@ func (tdb *TigrisDB) BuildFilter(filter *types.Document) driver.Filter {
 	res := map[string]any{}
 
 	for k, v := range filter.Map() {
-		// filter only by _id for now
-		if k != "_id" {
+		// don't pushdown empty strings yet
+		if v == "" {
 			continue
 		}
 
-		switch v.(type) {
-		case string:
-			// filtering by string values is complicated if the storage supports encodings, collations, etc,
-			// but Tigris does not support any of these
-		case types.ObjectID:
-			// filtering by ObjectID is always safe
-		default:
-			// skip other types for now
+		if k != "" {
+			// don't pushdown $comment and dot notation yet
+			if k[0] == '$' || types.NewPathFromString(k).Len() > 1 {
+				continue
+			}
+		}
+
+		// _id field supports only specific types
+		if k == "_id" {
+			switch v.(type) {
+			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
+			// type not supported for pushdown
+
+			case float64, string, types.ObjectID, int32, int64:
+				// filter by the exact _id value
+				id := must.NotFail(tjson.Marshal(v))
+				res["_id"] = json.RawMessage(id)
+			default:
+				panic(fmt.Sprintf("Unexpected type of field %s: %T", k, v))
+			}
+
 			continue
 		}
 
-		// filter by the exact _id value
-		id := must.NotFail(tjson.Marshal(v))
-		res["_id"] = json.RawMessage(id)
+		switch v.(type) {
```

Comment:
What is the difference between `k == "_id"` and other fields? Both switches look quite identical to me.

Reply:
Well, at the moment, we don't validate queries. If someone tries to set `_id` as an array, what will be the result? Nothing will be found, and it's an expected result. Why should we validate queries then?

Reply:
valid point, going to fix that